game-core/src/main/java/games/strategy/triplea/delegate/battle/casualty/AaCasualtySelector.java

```diff
@@ -114,265 +112,209 @@ private static CasualtyDetails getLowLuckAaCasualties(
+    final int guaranteeHitGroupSize =
+        calculateGuaranteeLowLuckHitGroupSize(targets, diceRoll, unitPowerAndRollsMap);
+
+    final Collection<UnitCategory> groupedTargets =
+        UnitSeparator.categorize(targets, null, false, true);
+    final List<List<Unit>> guaranteedHitGroups = new ArrayList<>();
+    final List<Unit> remainderUnits = new ArrayList<>();
+    for (final UnitCategory uc : groupedTargets) {
+      final Deque<List<Unit>> guaranteedGroups =
+          new ArrayDeque<>(Lists.partition(uc.getUnits(), guaranteeHitGroupSize));
+      final List<Unit> lastGroup = guaranteedGroups.peekLast();
+      // if the last group isn't the right size, put those units in the remainder list
+      if (lastGroup != null && lastGroup.size() != guaranteeHitGroupSize) {
+        guaranteedGroups.removeLast();
+        remainderUnits.addAll(lastGroup);
       }
-    } else if (hits < availableTargets.size()) {
-      // there isn't a roll for every target so need to randomly pick the target for each hit
-      final int[] hitRandom =
-          bridge.getRandom(
-              availableTargets.size(),
-              hits,
-              null,
-              DiceType.ENGINE,
-              "Deciding which planes should die due to AA fire");
-      // turn the random numbers into a unique set of targets
-      hitTargets = new HashSet<>();
-      int index = 0;
-      for (final int randomIndex : hitRandom) {
-        index = (index + randomIndex) % availableTargets.size();
-        while (hitTargets.contains(index)) {
-          index = (index + 1) % availableTargets.size();
-        }
-        hitTargets.add(index);
+      guaranteedHitGroups.addAll(guaranteedGroups);
+    }
+    return LowLuckTargetGroups.of(guaranteedHitGroups, remainderUnits);
+  }
+
+  @Value(staticConstructor = "of")
+  private static class LowLuckTargetGroups {
+    List<List<Unit>> guaranteedHitGroups;
+    List<Unit> remainderUnits;
+  }
+
+  /**
+   * Calculate the number of targets that guarantee a hit in a low luck dice roll
+   *
+   * <p>In low luck, the number of hits = (power / dice sides). If the strength for all of the aa
+   * units is the same, then the number of hits = (strength * targetCount / diceSides). To find out
+   * how big a group is needed to get a guaranteed hit, re-order the equation to be (targetCount /
+   * hits) = (diceSides / strength). So, for every (diceSides / strength) targets, there is one
+   * guaranteed hit.
+   *
+   * @return 0 if not possible to split up the targets into guaranteed hit groups or > 0 for the
+   *     size that guarantees a hit
+   */
+  private static int calculateGuaranteeLowLuckHitGroupSize(
+      final Collection<Unit> availableTargets,
+      final DiceRoll diceRoll,
+      final AaPowerStrengthAndRolls unitPowerAndRollsMap) {
+    final int bestStrength = unitPowerAndRollsMap.getBestStrength();
+    final int chosenDiceSize = unitPowerAndRollsMap.getDiceSides();
+
+    final boolean hasOverstackHits =
```

Comment:
It's not immediately clear what an overstack hit is. Maybe a quick comment explaining the term, or a different variable name would make this clear? I struggled here a bit and then reading a bit more later about the overstack AA hits it made more sense (but I'm still not sure I understand the concept TBH).